File: edu.wpi.first.wpilib.plugins.cpp/resources/templates/examples/PacGoat/src/Commands/OpenClaw.cpp
#include "OpenClaw.h"

#include "Robot.h"

OpenClaw::OpenClaw() {
	Requires(Robot::collector.get());
}

// Called just before this Command runs the first time
void OpenClaw::Initialize() {
	Robot::collector->Open();
}

// Make this return true when this Command no longer needs to run execute()
bool OpenClaw::IsFinished() {
	return Robot::collector->IsOpen();
}

File: edu.wpi.first.wpilib.plugins.java/resources/templates/examples/PacGoat/src/org/usfirst/frc/team190/pacgoat/commands/OpenClaw.java
package $package.commands;

import edu.wpi.first.wpilibj.command.Command;

import $package.Robot;

/**
 * Opens the claw
 */
public class OpenClaw extends Command {
	public OpenClaw() {
		requires(Robot.collector);
	}

	// Called just before this Command runs the first time
	@Override
	protected void initialize() {
		Robot.collector.open();
	}

	// Make this return true when this Command no longer needs to run execute()
	@Override
	protected boolean isFinished() {
		return Robot.collector.isOpen();
	}
}
